### PR TITLE
Add a linting container

### PR DESCRIPTION
**Pull request by maelk**

### CODE REVIEW

File: container/run-lint.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -ue
+
+SCRIPTPATH="$(dirname "$(readlink -f "${0}")")"
+
+docker build "${SCRIPTPATH}/lint" -t lint
+docker run -v "${SCRIPTPATH}"/../:/data --rm -it --name lint lint
```
Comment (kanwar-saad) [inline container/run-lint.sh]: why -it?

File: container/lint/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# (software-properties-common for add-apt-repository)
+RUN apt-get update && \
+    apt-get install -y \
+    git \
+    curl \
+    apt-transport-https \
+    ca-certificates \
+    software-properties-common
+
+#Add repos and install Go, Docker and Bazel
```
Comment (kanwar-saad) [inline container/lint/Dockerfile]: comment is wrong

File: container/lint/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    apt-get update && apt-get install -y \
+    golang-go \
+    golint \
+    shellcheck
+
+ENV GOPATH=/data/go
+ENV PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/data/go/bin
+
+
+#Install Golang dep, operator-sdk, Kubectl, Kustomize and kubebuilder
```
Comment (kanwar-saad) [inline container/lint/Dockerfile]: same

File: container/run-lint.sh (hunk 1 above)
Comment (kanwar-saad) [reply container/run-lint.sh]: i guess it should trigger a script which will just run all kind of linters on given folder. like go linters on all go files and python linter on .py files and same for .sh

**Comment by kanwar-saad** [comment]
test this please

**Comment by kanwar-saad** [comment]
retest

File: container/run-lint.sh (hunk 1 above)
Comment (maelk) [reply container/run-lint.sh]: About triggering a script, that's rather a use of the container than  the container itself. the run-lint.sh script is a convenience script to run it locally with a bash inside. If you want to automate the linting, then another script should be made IMO.

**Comment by maelk** [comment]
@mkth1 is implementing it using separated containers

**Pull request closed without merging**